topics/command-tips.md

```diff
@@ -0,0 +1,79 @@
+# Command tips
+
+An @array-reply of strings, command tips are meant to provide additional information about commands for clients and proxies.
+This infromation should help clients to determine how to handle this command when Redis is in clustering mode.
+Unlike the command flags (the third element in `COMMAND`'s reply), which are internal in nature (Redis itself uses them internally), command tips are just exposed via `COMMAND` and Redis does not use them.
+
+The tips are arbitrary strings, but in order to create common conventions, here is a list of suggested tips:
+
+- **blocking**
+- **random-output**
+- **random-order-output**
+- **block_keyword:<word>**
+- **request_policy:<policy>**
+- **reply_policy:<policy>**
+
+### blocking
+
+If present, this command has the potential to block (but not always, see `block_keyword` below)
+
+### random-output
+
+The output is not deterministic, that is, the same command with the same arguments, with the same key-space, may have different results. (e.g. `RANDOMKEY`, `SPOP`)
+
+### random-order-output
+
+The output is deterministic, but comes in random order (e.g. `HGETALL`, `SMEMBERS`)
+
+### block_keyword
+
+Identify commands which the potential to block, only is a specific arguement was given: For example, `XREAD` and `XREADGROUP` have the `blocking` tip, but they will only block if `BLOCK` was given. We will add the `block_keyword:<word>` tip, and the client should search `<word>` search in `argv` to determine if the command may indeed block.
+This could result in false-positive if that keyword is a keyname or some other form of free-text, but this tip best-effort: there's little harm in mistaking a non-blocking command with a blocking one (but mistaking in the other direction is unacceptable).
+
+### request_policy
+
+This tip should help clients to determine to which shard(s) to send the command in clustering mode.
+The default behavior (i.e. if the `request_policy` tip is absent) devides into two cases:
+1. The command has key(s). In this case the command goes to a single shard, determined by the hslot(s) of the key(s)
+2. The command doesn't have key(s). In this case the command can be sent to any arbitrary shard. In order to correctly process certain key-less commands, they should always go to the same shard (Example: Pub/Sub will not work properly if `SUBSCRIBE` and `PUBLISH` would reach different shards).
```

Comment:
Pubsub works "correctly" if the commands reach different shards.

Reply:
i'll remove the last sentence